backend/src/main/java/fk/prof/backend/aggregator/AggregationWindow.java

```diff
@@ -0,0 +1,159 @@
+package fk.prof.backend.aggregator;
+
+import fk.prof.backend.exception.AggregationFailure;
+import fk.prof.backend.model.request.RecordedProfileIndexes;
+import recording.Recorder;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AggregationWindow {
+  private String appId;
+  private String clusterId;
+  private String procId;
+
+  private final ConcurrentHashMap<Long, ProfileWorkInfo> workStatusLookup = new ConcurrentHashMap<>();
+  private LocalDateTime start = null, endWithTolerance = null;
```

Comment:
Discussed offline. We are sticking with localdatetime for now, and UTC clock across the board